app/code/community/Oyst/OneClick/Model/Payment/Method/Oneclick.php

```diff
@@ -42,4 +44,22 @@ public function getName()
     {
         return self::PAYMENT_METHOD_NAME;
     }
+
+    /**
+     * Refund specified amount for payment.
+     *
+     * @param Varien_Object $payment
+     * @param float $amount
+     *
+     * @return $this
+     */
+    public function refund(Varien_Object $payment, $amount)
+    {
+        /** @var $api Oyst_OneClick_Model_Payment_ApiWrapper */
+        $api = Mage::getModel('oyst_oneclick/payment_apiWrapper');
```

Comment:
In case of 1C we work with order, so you have to use the [Order API](https://github.com/oystparis/oyst-php/blob/b2ed807623d88c19335530281431589890a51b8f/src/Api/OystOrderApi.php#L123) ([details](https://github.com/oystparis/oyst-php/blob/b2ed807623d88c19335530281431589890a51b8f/src/Config/description_order.json#L57))

Reply:
for this you can use : app/code/community/Oyst/OneClick/Model/Order/ApiWrapper.php and implement the ```refunds()``` method